src/common/timing/time-keeper.js

```diff
@@ -95,6 +95,10 @@ export class TimeKeeper {
     return Math.floor(this.#correctedOriginTime + relativeTime)
   }
 
+  convertAbsoluteTimestamp (timestamp) {
+    return Math.floor(this.correctAbsoluteTimestamp(timestamp) - this.#correctedOriginTime)
+  }
```

Comment:
I think we should probably consider keeping this method more pure.  I would keep the correction part as separate call and have whomever needs both correction `and` a conversion to call them both separately.  The reason is that if you accidentally pass a corrected unix stamp into this, it will break it before transforming it.

Reply:
I am a little torn on this comment. I get where you are coming from but `convertRelativeTimestamp` already converts a relative timestamp to a corrected epoch timestamp. Don't you think it would be a little weird to have the opposite conversion be different? Maybe both the `convert` methods should convert based on the original origin time instead of the corrected origin time?